packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -11,6 +11,8 @@ import URI from '@theia/core/lib/common/uri';
 import { ILogger, DisposableCollection, Resource, Event, Emitter, ResourceProvider, MaybePromise } from "@theia/core/lib/common";
 import { PreferenceProvider } from '@theia/core/lib/browser/preferences';
 
+const detectIndent = require('detect-indent');
```

Comment:
please add it as an explicit dependency to `preferences` extension and install typings (https://www.npmjs.com/package/@types/detect-indent) to use es6 import syntax. Using es5 syntax is fine only if there is no another way.

Reply:
reworked to `jsoncparser`